covalent_dispatcher/_service/app_dask.py

```diff
@@ -0,0 +1,225 @@
+        return cluster_logs
+
+    async def _restart_worker(self, worker_id, worker_nanny_addr):
+        async with rpc(worker_nanny_addr) as r:
+            await r.restart()
+
+    async def _cluster_restart(self):
+        """
+        Restart the workers using their respective nanny services
+        """
+        worker_service_addrs = {}
+        async with rpc(self.cluster.scheduler_address) as r:
+            cinfo = await r.identity()
+            for _, worker_info in cinfo["workers"].items():
+                worker_id = worker_info["id"]
+                worker_service_addrs[f"{worker_id}"] = worker_info["nanny"]
+
+        await asyncio.gather(
+            *[
+                self._restart_worker(worker_id, nanny_addr)
+                for worker_id, nanny_addr in worker_service_addrs.items()
+            ]
+        )
+
+    async def _get_cluster_info(self):
+        """
+        Retrive cluster info from the scheduler
+        """
+        async with rpc(self.cluster.scheduler_address) as r:
+            return await r.identity()
+
+    async def _get_cluster_status(self):
+        """
+        Retrive status of the scheduler and all workers part of the cluster
+        """
+        proc_status = {}
+        async with rpc(self.cluster.scheduler_address) as r:
+            cinfo = await r.identity()
+
+        if cinfo:
+            proc_status["scheduler"] = "running"
+
+        for _, worker_info in cinfo["workers"].items():
+            worker_id = worker_info["id"]
+            proc_status[f"worker-{worker_id}"] = worker_info["status"]
+
+        return proc_status
+
+    async def _get_cluster_addresses(self):
+        """
+        Retrive the scheduler and worker addresses that are part of the cluster
+        """
+        addresses = {}
+        async with rpc(self.cluster.scheduler_address) as r:
+            cinfo = await r.identity()
+
+        addresses["scheduler"] = cinfo["address"]
+        addresses["workers"] = {}
+        for addr, worker_info in cinfo["workers"].items():
+            worker_id = worker_info["id"]
+            addresses["workers"][f"{worker_id}"] = addr
+
+        return addresses
+
+    def __len__(self) -> int:
+        """
+        Return the number of active workers in the cluster
+        """
+        if self.cluster:
+            return len(self.cluster.workers)
+        return 0
+
+    def run(self):
+        asyncio.set_event_loop(self.event_loop)
+        s = Server(handlers=self.handlers)
+        self.event_loop.create_task(s.listen(f"tcp://{self._admin_host}:{self._admin_port}"))
+
+
+class DaskCluster(Process):
+    """
+    Start a local Dask cluster in a separate multiprocessing process whose
+    lifetime is tied to Covalent. The cluster also starts and
+    admin/monitoring server along with it so that the cluster can be
+    administered and dynamically altered after covalent has started. By
+    default the admin server listens for TCP connections at
+    tcp://127.0.0.1:8000
```

Comment:
Is the admin server listening on `8000`? Below I see it's `get_random_available_port()`

Reply:
updated the doc string to reflect the actual behavior